astropy/cosmology/units.py

```diff
@@ -386,10 +386,9 @@ def with_H0(H0: Quantity | None = None) -> Equivalency:
 
     Parameters
     ----------
-    H0 : None or `~astropy.units.Quantity` ['frequency']
-        The value of the Hubble constant to assume. If a
-        `~astropy.units.Quantity`, will assume the quantity *is* ``H0``. If
-        `None` (default), use the ``H0`` attribute from
+    H0 : None or Quantity['frequency']
```

Comment:
Here you don't have a space between `Quantity` and `['...']` - I think that is clearer - do it everywhere? Or insert space here for consistency.

Reply:
I think (but am not 100% sure) that the space is needed for numpydoc xref
subsitution + the annotation tool to work together correctly.